### PR TITLE
fix: replace defines with constexpr/const and use more absolute paths for includes

### PR DESCRIPTION
**Pull request by Nerixyz**
- Replaces constant `#define`s with `constexpr` where possible. In case a `QString` is desired, they're replaced with a static `QString`.
	- For concatenations, QStringBuilder is required. This is included with QString in Qt 6 but requires an include in Qt 5
- Uses paths relative to `src/` for all includes. Check with `rg -P '^#include "(?!Application|BrowserExtension|RunGui|ForwardDecl)[^"/]+"' -g 'src/**/*.{cpp,hpp}'`.

### CODE REVIEW

File: src/common/Credentials.cpp [hunk 1]
```diff
@@ -11,31 +11,24 @@
...
 #include <variant>
 
 #ifndef NO_QTKEYCHAIN
-#    ifdef CMAKE_BUILD
-#        if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
-#            include "qt6keychain/keychain.h"
-#        else
-#            include "qt5keychain/keychain.h"
-#        endif
-#    else
```
Comment (pajlada) [inline src/common/Credentials.cpp:24]: this breaks my build

File: src/common/Version.cpp [hunk 2]
```diff
@@ -10,8 +10,31 @@ namespace chatterino {
...
+ *  - 2.4.0-beta.3
+ *  - 2.4.0-beta.2
+ *  - 2.4.0-beta
+ *
+ * Alpha:
+ *  - 2.4.0-alpha.3
+ *  - 2.4.0-alpha.2
+ *  - 2.4.0-alpha
+ **/
+constexpr QStringView CHATTERINO_VERSION = u"2.5.1";
```
Comment (pajlada) [inline src/common/Version.cpp:34]: Why not keep this defined in the header and continue to use `CHATTERINO_VERSION` everywhere?

File: src/widgets/helper/CommonTexts.hpp [hunk 3]
```diff
@@ -1,8 +1,13 @@
...
 
-#define OPEN_IN_BROWSER "Open stream in browser"
-#define OPEN_PLAYER_IN_BROWSER "Open player in browser"
-#define OPEN_MOD_VIEW_IN_BROWSER "Open mod view in browser"
-#define OPEN_IN_STREAMLINK "Open in streamlink"
-#define DONT_OPEN "Don't open"
-#define OPEN_WHISPERS_IN_BROWSER "Open whispers in browser"
+#include <QString>
+
+inline const QString OPEN_IN_BROWSER = QStringLiteral("Open stream in browser");
```
Comment (pajlada) [inline src/widgets/helper/CommonTexts.hpp:5]: do we avoid using the `u"..."_s` syntax here to keep the header file simpler?

File: src/widgets/helper/NotebookButton.cpp [hunk 4]
```diff
@@ -14,8 +14,6 @@
 #include <QPainterPath>
 #include <QRadialGradient>
 
-#define nuuls nullptr
```
Comment (pajlada) [inline src/widgets/helper/NotebookButton.cpp:17]: ?

File: src/widgets/settingspages/IgnoresPage.cpp [hunk 5]
```diff
@@ -19,12 +20,14 @@
 #include <QTableView>
 #include <QVBoxLayout>
 
+namespace chatterino {
+
+using namespace literals;
+
 // clang-format off
-#define INFO "/block <user> in chat blocks a user.\n/unblock <user> in chat unblocks a user.\nYou can also click on a user to open the usercard."
+inline const QString INFO = u"/block <user> in chat blocks a user.\n/unblock <user> in chat unblocks a user.\nYou can also click on a user to open the usercard."_s;
```
Comment (pajlada) [inline src/widgets/settingspages/IgnoresPage.cpp:28]: Can just inline this in `addUsersTab` instead, like you've done in HighlightingPage

**Review by pajlada — COMMENTED** [review]
small incomplete review

File: src/common/Version.cpp (hunk 2 above)
Comment (Nerixyz) [reply src/common/Version.cpp]: Done.

File: src/widgets/helper/CommonTexts.hpp (hunk 3 above)
Comment (Nerixyz) [reply src/widgets/helper/CommonTexts.hpp:7]: We can't use it in header files, as we'd need to add a `using` directive - effectively making the operator `""_s`  available everywhere.

File: src/widgets/helper/NotebookButton.cpp (hunk 4 above)
Comment (Nerixyz) [reply src/widgets/helper/NotebookButton.cpp:17]: This was never used.

Comment (pajlada) [reply src/widgets/helper/NotebookButton.cpp:17]: sad

Comment (Nerixyz) [reply src/widgets/helper/NotebookButton.cpp:17]: I mean I can add it back

Comment (pajlada) [reply src/widgets/helper/NotebookButton.cpp:17]: No it's ok, just end of an era

Comment (Mm2PL) [reply src/widgets/helper/NotebookButton.cpp:17]: sadge

**Review by pajlada — APPROVED** [review]

**Pull request merged**